### ISSUE TITLE
Bug: pnpm exec eslint didn't work on Mac os

### ISSUE THREAD

**Issue by zhanglolo** [eslint/eslint#16257]
### Environment

Environment Info:

Node version: v16.17.0
npm version: v8.15.0
pnpm version: v7.9.5
Local ESLint version: v8.23.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.6.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
//eslint.config.js
export default [
  {
    rules: {
      semi: "warn",
      "prefer-const": "error",
    },
  },
];
```
</details>

```js
//Missing semicolon
Vue.use(PiniaVuePlugin)
```

### What did you expect to happen?

config script in package.json
```json
"lint": "eslint src/**/*.js",
```

run `pnpm run lint ` terminal print nothing.

run `pnpm exec eslint src/**/*.js`, output:
```shell

/Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js
  9:24  warning  Missing semicolon  semi

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

### What actually happened?

It seem like eslint didn't work by using package.json script.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by kecrily** [comment]
Unable to reproduce

![image](https://user-images.githubusercontent.com/45708948/187140953-343554ec-a280-4231-b2ad-9650fca3cf68.png)

**Comment by zhanglolo** [comment]
@kecrily Windows look fine.  May be just Mac OS 12.5.1 happend. Need other Mac user feedback.

<img width="985" alt="截屏2022-08-29 15 03 09" src="https://user-images.githubusercontent.com/27875371/187142859-2a9a08de-3846-4180-8abf-b9278e7e0cc9.png">

**Comment by zhanglolo** [comment]
It works after reopen vscode.

**Comment by zhanglolo** [comment]
> Unable to reproduce in Windows
> 
> ![image](https://user-images.githubusercontent.com/45708948/187140953-343554ec-a280-4231-b2ad-9650fca3cf68.png)

did you config `eslint.config.js` like this:
```js
//eslint.config.js
export default [
  {
    rules: {
      semi: "warn",
      "prefer-const": "error",
    },
  },
];
```

**Comment by kecrily** [comment]
Yes, I copied it.

**Comment by ljharb** [comment]
If something works with npm but breaks with pnpm, then you may need to file an issue on pnpm.

**Comment by mdjermanovic** [comment]
@zhanglolo thanks for the feedback on the new config system!

> ```json
> "lint": "eslint src/**/*.js",
> ```

Can you add the [`--debug`](https://eslint.org/docs/latest/user-guide/command-line-interface#--debug) flag here, run the script again, and share the output?

**Comment by zhanglolo** [comment]
According to debug information, actually, this issue is `pnpm exec eslint src/**/*.js` did find `src/main.js` but `pnpm run lint`  or `npx run lint` didn't.

run `pnpm exec eslint src/**/*.js` in terminal and `"lint": "eslint src/**/*.js` in package.json find different files which need to handle.

## Detail information:
`pnpm run lint ` :
```shell
 *  正在执行任务: pnpm run lint 


> vue2-vite-pinia-vitest-demo@0.0.0 lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo
> eslint src/**/*.js --debug

  eslint:cli CLI args: [ 'src/api/index.js', 'src/mock/browser.js', 'src/mock/handlers.js', 'src/mock/server.js', 'src/router/index.js', 'src/store/counter.js', 'src/store/index.js', '--debug' ] +0ms
  eslint:cli Using flat config? true +2ms
  eslint:cli Running on files +3ms
  eslint:flat-eslint Searching for eslint.config.js +0ms
  eslint:flat-eslint Loading config from /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/eslint.config.js +0ms
  eslint:flat-eslint Config file URL is file:///Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/eslint.config.js +0ms
  eslint:flat-eslint Deleting cache file at /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/.eslintcache +4ms
  eslint:rules Loading rule 'semi' (remaining=286) +0ms
  eslint:rules Loading rule 'prefer-const' (remaining=285) +5ms
  eslint:flat-eslint 7 files found in: 9ms +9ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js (pass 1) +4ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +6ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +5ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +1ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js (pass 1) +3ms
  eslint:source-code-fixer Applying fixes +9ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +9ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +1ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +2ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +2ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +1ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +1ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js (pass 1) +1ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +1ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +1ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js (pass 1) +1ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +1ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +1ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +3ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms

/Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js
  12:21  warning  Missing semicolon  semi

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

run `npm run lint`:
```shell
❯ npm run lint         

> vue2-vite-pinia-vitest-demo@0.0.0 lint
> eslint src/**/*.js --debug

  eslint:cli CLI args: [ 'src/api/index.js', 'src/mock/browser.js', 'src/mock/handlers.js', 'src/mock/server.js', 'src/router/index.js', 'src/store/counter.js', 'src/store/index.js', '--debug' ] +0ms
  eslint:cli Using flat config? true +3ms
  eslint:cli Running on files +2ms
  eslint:flat-eslint Searching for eslint.config.js +0ms
  eslint:flat-eslint Loading config from /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/eslint.config.js +0ms
  eslint:flat-eslint Config file URL is file:///Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/eslint.config.js +0ms
  eslint:flat-eslint Deleting cache file at /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/.eslintcache +4ms
  eslint:rules Loading rule 'semi' (remaining=286) +0ms
  eslint:rules Loading rule 'prefer-const' (remaining=285) +7ms
  eslint:flat-eslint 7 files found in: 14ms +14ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js (pass 1) +3ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +6ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js (pass 1) +1ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +2ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +1ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +4ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +4ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +3ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +1ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js (pass 1) +2ms
  eslint:source-code-fixer Applying fixes +6ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +6ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +1ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +1ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +1ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +1ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +1ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +3ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +3ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +1ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +2ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms

/Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js
  12:21  warning  Missing semicolon  semi

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```

both find 7 js files in src:
 `'src/api/index.js', 'src/mock/browser.js', 'src/mock/handlers.js', 'src/mock/server.js', 'src/router/index.js', 'src/store/counter.js', 'src/store/index.js'`

and there was a warning  Missing semicolon  semi in src/store/counter.js.

---
run `pnpm exec eslint src/**/*.js`:
```shell
❯ pnpm exec eslint src/**/*.js --debug  
  eslint:cli CLI args: [ 'src/api/index.js', 'src/main.js', 'src/mock/browser.js', 'src/mock/handlers.js', 'src/mock/server.js', 'src/router/index.js', 'src/store/counter.js', 'src/store/index.js', '--debug' ] +0ms
  eslint:cli Using flat config? true +2ms
  eslint:cli Running on files +3ms
  eslint:flat-eslint Searching for eslint.config.js +0ms
  eslint:flat-eslint Loading config from /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/eslint.config.js +1ms
  eslint:flat-eslint Config file URL is file:///Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/eslint.config.js +0ms
  eslint:flat-eslint Deleting cache file at /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/.eslintcache +4ms
  eslint:rules Loading rule 'semi' (remaining=286) +0ms
  eslint:rules Loading rule 'prefer-const' (remaining=285) +7ms
  eslint:flat-eslint 8 files found in: 13ms +13ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +1ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/api/index.js (pass 1) +4ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js +6ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js +4ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js +1ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js (pass 1) +2ms
  eslint:source-code-fixer Applying fixes +7ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +7ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +2ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js +1ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/router/index.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +4ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +4ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +2ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/handlers.js (pass 1) +2ms
  eslint:source-code-fixer Applying fixes +4ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +5ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js (pass 1) +1ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/server.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +2ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +1ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/mock/browser.js (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js (pass 1) +1ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js (pass 1) +1ms
  eslint:source-code-fixer Applying fixes +2ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:flat-eslint Lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +2ms
  eslint:linter Linting code for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Parsing: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Parsing successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Scope analysis: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Scope analysis successful: /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js +0ms
  eslint:linter Generating fixed text for /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/index.js (pass 1) +0ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms

/Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/main.js
  9:24  warning  Missing semicolon  semi

/Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo/src/store/counter.js
  12:21  warning  Missing semicolon  semi

✖ 2 problems (0 errors, 2 warnings)
  0 errors and 2 warnings potentially fixable with the `--fix` option.
```
8 files, did find `src/main.js`.
so there was 2 warnings. (in `src/main.js` and `src/store/counter.js`)

**Comment by mdjermanovic** [comment]
> ```shell
>  *  正在执行任务: pnpm run lint 
> 
> 
> > vue2-vite-pinia-vitest-demo@0.0.0 lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo
> > eslint src/**/*.js --debug
> 
>   eslint:cli CLI args: [ 'src/api/index.js', 'src/mock/browser.js', 'src/mock/handlers.js', 'src/mock/server.js', 'src/router/index.js', 'src/store/counter.js', 'src/store/index.js', '--debug' ] +0ms
> ```

> run `pnpm exec eslint src/**/*.js`:
> 
> ```shell
> ❯ pnpm exec eslint src/**/*.js --debug  
>   eslint:cli CLI args: [ 'src/api/index.js', 'src/main.js', 'src/mock/browser.js', 'src/mock/handlers.js', 'src/mock/server.js', 'src/router/index.js', 'src/store/counter.js', 'src/store/index.js', '--debug' ] +0ms
> ```

Thanks for the info!

In both of these cases, before calling eslint the glob was expanded to a list of file paths which were then passed to eslint as individual cli arguments. I'm not sure why the lists are different, but either way this happens outside eslint and eslint has no control over it.

Can you try with quotes?

```
"lint": "eslint \"src/**/*.js\"",
```

This way, the glob should be passed to eslint as-is, and eslint itself will resolve it. (to doublecheck, with the `--debug` flag the original glob pattern should appear in the `eslint:cli CLI args: [...]` line instead of the file paths).

**Comment by zhanglolo** [comment]
@mdjermanovic modify package.json  `"lint": "eslint \"src/**/*.js\" --debug"`

`pnpm run lint`
```shell
 *  正在执行任务: pnpm run lint 


> vue2-vite-pinia-vitest-demo@0.0.0 lint /Users/zhanglolo/Documents/vscodeProjects/vue2-vite-pinia-vitest-demo
> eslint "src/**/*.js" --debug

  eslint:cli CLI args: [ 'src/**/*.js', '--debug' ] +0ms
```

`pnpm exec eslint "src/**/*.js`:
```shell
❯ pnpm exec eslint "src/**/*.js" --debug  
  eslint:cli CLI args: [ 'src/**/*.js', '--debug' ] +0ms
```

`npx eslint "src/**/*.js"`:
```shell
❯ npx eslint "src/**/*.js" --debug
  eslint:cli CLI args: [ 'src/**/*.js', '--debug' ] +0ms
```

**Comment by mdjermanovic** [comment]
Does it work as expected now (i.e. does it report the `semi` warning in `src/main.js` when you run the package.json `lint` script)?

**Comment by zhanglolo** [comment]
> Does it work as expected now (i.e. does it report the `semi` warning in `src/main.js` when you run the package.json `lint` script)?

Yes! 
but, what makes me confuses is `pnpm run lint ` (`npm run lint`) get different  out put `eslint:cli CLI args: [...]`  from `pnpm exec eslint src/**/*.js`(`npx eslint src/**/*.js`). 

May be npm have do something stuff.🤨

**Comment by mdjermanovic** [comment]
> > Does it work as expected now (i.e. does it report the `semi` warning in `src/main.js` when you run the package.json `lint` script)?
> 
> Yes! but, what makes me confuses is `pnpm run lint ` (`npm run lint`) get different out put `eslint:cli CLI args: [...]` from `pnpm exec eslint src/**/*.js`(`npx eslint src/**/*.js`).
> 

Then I think we can close this issue since eslint correctly resolves the glob when the glob is passed into eslint. The difference between the lists of files to which the glob is expanded before calling eslint might be a good question for npm/pnpm, as eslint has no control over it. ESLint's [Command Line Interface](https://eslint.org/docs/latest/user-guide/command-line-interface) documentation suggests using quotes for consistent behavior:

> Please note that when passing a glob as a parameter, it will be expanded by your shell. The results of the expansion can vary depending on your shell, and its configuration. If you want to use node glob syntax, you have to quote your parameter (using double quotes if you need it to run in Windows), as follows
